Arch arm float configurable mpu guards (#17034)
zephyrproject-rtos/zephyr: doc/reference/kernel/other/float.rst

The ARM kernel automatically detects that a given thread is using the floating
point registers the first time the thread accesses them.

> Pretag a thread that intends to use the FP registers by

Reviewer: You commented that this pretagging is optional, so you **could** write this instead as:
```
You can optionally pretag a thread that intends ...
```

> It is recommended to pretag a thread that intends to use the FP registers,

Reviewer: This would read a bit better
```suggestion
Pretag a thread that intends to use the FP registers by
```

Author: Will do that too, thanks.

Author: @dbkinder I just want to make sure that we don't give the impression that pre-tagging is (absolutely) required for FP to work. That's why I added the "recommended". I've changed to what you suggested, pls, ACK

using one of the techniques listed below.

* A statically-created ARM thread can be pretagged by passing the

>   :c:macro:`K_FP_REGS`option to :c:macro:`K_THREAD_DEFINE`.

Reviewer: ```suggestion
  :c:macro:`K_FP_REGS` option to :c:macro:`K_THREAD_DEFINE`.
```

Author: Fixing that, thanks.
